### PR TITLE
223150: AKS Internal Loadbalancer

### PR DESCRIPTION
**Pull request by ppotturi**
# Description
Install internal loadbalancer and change the outbound traffic to UDR.

[AB#223150](https://dev.azure.com/defragovuk/93c65135-d16b-49d6-a191-4a5313532779/_workitems/edit/223150)

# Testing
Successfully deployment with the pipeline.

### CODE REVIEW

File: infra/managed-cluster/.bicep/acr-pull.bicep [hunk 1]
```diff
@@ -0,0 +1,18 @@
+@description('Required. The parameter object for the managed identity. The object must contain the name and principalId values.')
+param principalId string
+@description('Required. The name of the vnet')
```
Comment (ppotturi) [inline infra/managed-cluster/.bicep/acr-pull.bicep:3]: :-)

**Review by ajaysoni-cap — DISMISSED** [review]

File: infra/managed-cluster/aks-cluster.bicep [hunk 2]
```diff
@@ -147,12 +149,12 @@ module deployAKS 'br/SharedDefraRegistry:container-service.managed-clusters:0.5.
     aksClusterNetworkPlugin: 'azure'
     aksClusterNetworkPluginMode: 'overlay'
     aksClusterNetworkPolicy: 'calico'
-    aksClusterPodCidr: '172.16.0.0/16'
-    aksClusterServiceCidr: '172.18.0.0/16'
+    aksClusterPodCidr: cluster.podCidr
+    aksClusterServiceCidr: cluster.serviceCidr
     aksClusterDnsServiceIP: '172.18.255.250'
```
Comment (DaRoszko) [inline infra/managed-cluster/aks-cluster.bicep:154]: where did this come from? but worthwhile wacking in common?

Comment (ppotturi) [reply infra/managed-cluster/aks-cluster.bicep:154]: This is from aks baseline. I will move it to vars.

**Review by kbabigumira — APPROVED** [review]

File: infra/network/virtual-network.parameters.json [hunk 3]
```diff
@@ -22,9 +22,10 @@
             { "service": "Microsoft.ServiceBus" },
             { "service": "Microsoft.KeyVault" }
           ],
+          "routeTableId": "/subscriptions/#{{ subscriptionId }}/resourceGroups/#{{ virtualNetworkResourceGroup }}/providers/Microsoft.Network/routeTables/UDR-Spoke-Route-From-#{{ networkResourceNamePrefix }}#{{ nc_resource_virtualnetwork }}#{{ nc_instance_regionid }}01",
           "networkSecurityGroupId":"/subscriptions/#{{ subscriptionId }}/resourceGroups/#{{ virtualNetworkResourceGroup }}/providers/Microsoft.Network/networkSecurityGroups/#{{ networkResourceNamePrefix }}#{{ nc_resource_nsg }}#{{ nc_instance_regionid }}01",
           "privateEndpointNetworkPolicies": "Disabled",
-          "privateLinkServiceNetworkPolicies": "Enabled"
+          "privateLinkServiceNetworkPolicies": "Disabled"
```
[the remark this answers is not in the document]
Comment (DaRoszko) [reply infra/network/virtual-network.parameters.json:28]: u got a link to that?

**Pull request merged**